deepcopy.go

```diff
@@ -132,6 +132,9 @@ func _map(x interface{}, ptrs map[uintptr]interface{}) (interface{}, error) {
 	if v.Kind() != Map {
 		return nil, fmt.Errorf("must pass a value with kind of Map; got %v", v.Kind())
 	}
+	if v.IsNil() {
```

Comment:
@mnovak-veza did you attempt this before? I vaguely remember it coming up before.

Reply:
The change is good.
Note that in some cases, where not possible otherwise, the tests that will be affected by the change will have to be modified (pass `cmpopts.EquateEmpty()` in assertutils compare) FYI @ajelic-cookieai

Reply:
Thanks for the tip @mnovak-veza, I'll keep an eye on the failing tests in my other PR: https://github.com/cookieai-jar/cookieai-core/pull/26884